interface/modules/custom_modules/oe-module-faxsms/src/Controller/ClickatellSMSClient.php

```diff
@@ -61,9 +45,9 @@ public function sendSMS($toPhone = '', string $subject = '', string $message = '
 
         $url = sprintf(
             "https://platform.clickatell.com/messages/http/send?apiKey=%s&to=%s&from=%s&content=%s",
-            $this->appKey,
+            $this->credentials['appKey'],
```

Comment:
Just a thought. You may want to validate credentials first. Could return empty.

Reply:
Good thought.  I'd lean towards keeping that check as close as possible to the user entering it.  Right now, the input fields on the service setup form (in `setup.php`) have a `required` attribute, so that validation will happen when the form is submitted.  In the case that the credentials are edited some other way, the API call will fail with an error message that will get exposed to the user.

As for checking that `$this->credentials` is actually set, since we're setting that in the constructor, the only way it could be null is if we made some logic error, which a user couldn't do anything about.  The send will fail one way or another.

My two cents would be to not add a check here since null `credentials` isn't an expected/possible condition.

Reply:
That's fine. It'd be very rare for me to try to force any change on you. So in future you don't have to justification just say you disagree or prefer to do as you have it.
Too be honest, I'm horrible at reviews and avoid them when I can so take that into account:)
If I feel strongly about something, I'll let you know. You're doing a great job and we're lucky to have you.